metadata_service/proxy/gremlin_proxy.py

```diff
@@ -90,115 +929,1004 @@ def _submit(self, *, command: str, bindings: Any = None) -> Any:
+
+        :param table_uri:
+        :param column_name:
+        :return:
+        """
+        column_uri = make_column_uri(table_uri=table_uri, column_name=column_name)
+        g = _V(g=self.g, label=VertexTypes.Column, key=column_uri)
+        g = g.outE(EdgeTypes.Description.value.label).inV()
+        g = g.has(VertexTypes.Description.value.label, 'source', 'user').values('description')
+        return self.query_executor()(query=g, get=FromResultSet.getOptional)
 
+    @timer_with_counter
+    @overrides
+    def get_popular_tables(self, *, num_entries: int = 10) -> List[PopularTable]:
+        """
+        Retrieve popular tables. As popular table computation requires full scan of table and user relationship,
+        it will utilize cached method _get_popular_tables_uris.
+
+        :param num_entries:
+        :return: Iterable of PopularTable
+        """
+
+        table_uris = self._get_popular_tables_uris(num_entries)
+        if not table_uris:
+            return []
+
+        g = _V(g=self.g, label=VertexTypes.Table, key=within(*table_uris), key_property_name=self.key_property_name)
+        g = g.as_('table')
+        g = g.inE(EdgeTypes.Table.value.label). \
+            outV().hasLabel(VertexTypes.Schema.value.label).as_('schema')
+        g = g.inE(EdgeTypes.Schema.value.label). \
+            outV().hasLabel(VertexTypes.Cluster.value.label).as_('cluster')
+        g = g.inE(EdgeTypes.Cluster.value.label). \
+            outV().hasLabel(VertexTypes.Database.value.label).as_('database')
+        g = g.coalesce(select('table').outE(EdgeTypes.Description.value.label).
+                       inV().has(VertexTypes.Description.value.label, 'source', 'user').fold()).as_('description')
+        g = g.select('database', 'cluster', 'schema', 'table', 'description'). \
+            by('name').by('name').by('name').by('name').by(unfold().values('description').fold())
+        results = self.query_executor()(query=g, get=FromResultSet.toList)
+
+        if not results:
+            return []
+
+        popular_tables = []
+        for result in results:
+            popular_tables.append(PopularTable(
+                database=result['database'], cluster=result['cluster'], schema=result['schema'], name=result['table'],
+                description=_safe_get(result, 'description')
+            ))
+        return popular_tables
+
+    # @timer_with_counter
+    # @_CACHE.cache('_get_popular_tables_uris', _GET_POPULAR_TABLE_CACHE_EXPIRY_SEC)
+    def _get_popular_tables_uris(self, num_entries: int) -> List[str]:
+        """
+        Retrieve popular table uris. Will provide tables with top x popularity score.
+        Popularity score = number of distinct readers * log(total number of reads)
+        The result of this method will be cached based on the key (num_entries), and the cache will be expired based on
+        _GET_POPULAR_TABLE_CACHE_EXPIRY_SEC
```

Comment:
this comment is a bit outdated -- we removed the cache from our gremlin impl

Reply:
Got it, removing the comment and commented out cache lines.